charts/kargo/templates/_helpers.tpl

```diff
@@ -6,40 +6,6 @@ Expand the name of the chart.
 {{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
-{{/*
-Create a default fully qualified app name.
-We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
-If release name contains chart name it will be used as a full name.
-*/}}
-{{- define "kargo.fullname" -}}
-{{- $name := default .Chart.Name .Values.nameOverride -}}
-{{- if contains $name .Release.Name -}}
-{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
-{{- else -}}
-{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
-{{- end -}}
-
-{{- define "kargo.api.fullname" -}}
-{{ include "kargo.fullname" . | printf "%s-api" }}
-{{- end -}}
-
-{{- define "kargo.controller.fullname" -}}
-{{ include "kargo.fullname" . | printf "%s-controller" }}
-{{- end -}}
-
-{{- define "kargo.dexServer.fullname" -}}
-{{ include "kargo.fullname" . | printf "%s-dex-server" }}
-{{- end -}}
-
-{{- define "kargo.garbageCollector.fullname" -}}
-{{ include "kargo.fullname" . | printf "%s-garbage-collector" }}
-{{- end -}}
-
-{{- define "kargo.webhooksServer.fullname" -}}
-{{ include "kargo.fullname" . | printf "%s-webhooks-server" }}
-{{- end -}}
-
```

Comment:
I agree with making the names less dynamic, for all the reasons you stated, _however_, I would have a slight preference for keeping these helpers and continuing to use them, but merely changing the implementation of `kargo.fullname` to be substantially simpler.

My reasoning for this is the same reason that one might prefer constants over repetitious use of a string literal in just about any programming language. When you fat finger a constant, you end up with an error. When you fat finger a string literal, you end up with a bug.

wdyt?

Reply:
I do see your point, but there was one other reason I neglected to mention for making this change: which was to improve the readability of the charts. Using the string literals, it:
* becomes much more clear on the resource references since it removes all ambiguity that might be obfuscated by the template function.
* removes a ton of `text/template` from the files and makes it look a lot more like kubernetes manifests

The template functions require one more level of mental indirection to the `_helpers.tpl` to understand the final output.

I'll use the counter example of how we also don't go to the lengths of variablizing container ports, even though a typo would cause us to suffer from a similar bug.

Reply:
I cannot deny it's more readable. I can live with that.